import/src/main/java/com/google/cloud/healthcare/imaging/dicomadapter/CMoveService.java

```diff
@@ -231,20 +231,16 @@ private void sendErrorResponse(int status, String message, List<String> failedIn
           MonitoringService.addEvent(Event.CMOVE_ERROR);
       }
 
-      if (message == null && (failedInstanceUids == null || failedInstanceUids.size() == 0)) {
-        as.tryWriteDimseRSP(pc, Commands.mkCMoveRSP(cmd, status));
-        return;
-      }
-
-      Attributes attributes = new Attributes();
+      Attributes cmdAttr = Commands.mkCMoveRSP(cmd, status);
+      Attributes dataAttr = new Attributes();
       if (message != null) {
-        attributes.setString(Tag.ErrorComment, VR.LO, message);
+        cmdAttr.setString(Tag.ErrorComment, VR.LO, message);
       }
       if (failedInstanceUids != null && failedInstanceUids.size() > 0) {
-        attributes.setString(Tag.FailedSOPInstanceUIDList, VR.UI,
+        dataAttr.setString(Tag.FailedSOPInstanceUIDList, VR.UI,
             failedInstanceUids.toArray(new String[]{}));
       }
-      as.tryWriteDimseRSP(pc, Commands.mkCMoveRSP(cmd, status), attributes);
+      as.tryWriteDimseRSP(pc, cmdAttr, dataAttr);
```

Comment:
Due to the way dcm4che is implemented, tryWriteDimseRSP calls into writeDimseRSP which when the dataset parameter is not null, sets CommandDataSetType to "with dataset" type instead of "no dataset" type. However there is no dataset. So we should avoid passing a dataset with 0 objects into tryWriteDimseRSP

Reply:
Nice find! Fixed.